opm/input/eclipse/Schedule/SourceProp.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright 2023 Equinor ASA.
+  Copyright 2023 Norce.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_SOURCE_PROP_HPP
+#define OPM_SOURCE_PROP_HPP
+
+#include <vector>
+#include <cstddef>
+#include <optional>
+
+#include <opm/input/eclipse/EclipseState/Grid/GridDims.hpp>
+
+namespace Opm {
+
+class Deck;
+class DeckRecord;
+
+enum class SourceComponent {
+     OIL,
+     GAS,
+     WATER,
+     SOLVENT,
+     POLYMER,
+     NONE
+};
+
+class SourceProp {
+public:
+
+    struct SourceCell {
+        int i, j, k;
```

Comment:
Do you really need to refer to the individual `i`, `j`, and `k` coordinates?  If not, maybe we could collect these into a `std::array<int,3>` instead?  That would also simplify the comparison operators.